### DISCUSSION TITLE
Installing TopoPyScale (issues with topocalc package)

### DISCUSSION THREAD

**Discussion by kyung-robin-kim** [ArcticSnow/TopoPyScale#108] [Q&A] [answered]
Hello, I'm trying to use your tool to downscale some ERA5 outputs, however, I'm running into issues with downloading toposcale. Would you know if there's an alternative way to download that package (or substitute package that's available)? I've tried using the .yml file, and calling it independently with -c conda-forge, pip, or pip3, but a common error appears: "ERROR: Failed building wheel for topocalc...Could not build wheels for topocalc, which is required to install pyproject.toml-based projects." "This error originates from a subprocess, and is likely not a problem with pip." Every other package has been able to get installed by following the guide online. Thank you.

**Comment by ArcticSnow** [comment]
Hello @kyung-robin-kim 

Can you try installing it with pip? It seems topocalc is giving problem: https://github.com/USDA-ARS-NWRC/topocalc

```sh
conda create -n downscaling python=3.9 ipython
conda activate downscaling

# Recomended way to install dependencies:
conda install -c conda-forge xarray matplotlib scikit-learn pandas numpy netcdf4 h5netcdf rasterio pyproj dask geopandas
pip install topocalc

# and finally
pip install topopyscale
```

**Reply by kyung-robin-kim** [reply]
Hello @ArcticSnow
Thank you for the quick reply; yes, I've tried to create a new environment and download as you suggested, but the error is persistent. I contacted Dr. Hendrick re: topocalc so I'll report back if I receive any updates/recommendations.

**Reply by ArcticSnow** [reply]
I was able to repdrouce the error too using a fresh VE. The problem for me was that numpy was missing

**Reply by kyung-robin-kim** [reply, marked as answer]
I recently heard back from Andrew, and he helped troubleshoot my specific error, which had to do with the parent virtual environment! I had been using "Miniconda" instead of the full Anaconda suite to minimize some packages. And using pip installed topocalc separately finally worked.

**Reply by ArcticSnow** [reply]
I also used Miniconda. Ok then very good to know. Thank you for sharing.
I hope you'll enjoy TopoPyScale then. If you find improvements to do feel free to propose PR.

**Reply by Hedrick-ARS** [reply]
Hello,

Andrew Hedrick here. Thank you Robin for bringing this compile issue to our attention. After your email yesterday I remembered that we have recently found that some Apple computers using the newer M chips have issues with some of the c-code within topocalc. So I do not think it is as simple as a Miniconda/Anaconda issue. 

Just food for thought. We hope to eventually fix this as we move our model framework into a more supported Conda environment.

Cheers!

**Reply by kyung-robin-kim** [reply]
Hello, I have some follow-up questions after trying to reinstall TopoPyScale on a new work computer (twice)...I used Anaconda, then  downloaded TopoPyScale using the .yaml file in https://github.com/ArcticSnow/TopoPyScale_examples. But unlike last time, I encountered some errors with the "git" package for some reason. Are you aware of any errors regarding git wrt TopoPyScale or its dependencies?

Also, could the yaml file be updated to include geopandas? Thank you!

**Reply by BINGZIEMMA** [reply]
Hello, I have encountered another problem. When I install topocalc with "pip install topocalc", the error "ERROR: Failed to build installable wheels for some pyproject.toml based projects (topocalc)" ocurred. Do you know how to sovle it? Thank you!

**Reply by ArcticSnow** [reply]
Topocalc is now ported to Python 3.13 which should prevent many incompabilities we recently encountered during installing. Use the forked version: https://github.com/ArcticSnow/topocalc

**Comment by ArcticSnow** [comment]
Problem solved
